frontend/passkeys.js

```diff
@@ -58,6 +60,53 @@ export async function authenticatePasskey(targetHref, csrfToken) {
     }
 }
 
+export function deletePasskey(passkeyName, csrfToken) {
+    try {
+        if (confirm(`Are you sure you want to delete the passkey "${passkeyName}"?`)) {
+            createAndSubmitForm('/passkey/delete', {
+                passkeyName: passkeyName,
+                csrfToken: csrfToken
+            });
+        }
+    } catch (err) {
+        console.error('Error during passkey deletion:', err);
+        M.toast({ html: 'Failed to delete passkey. Please try again.', classes: 'rounded red' });
+    }
+}
+
+export function setUpDeletePasskeyButtons(selector) {
+    const deleteButtons = document.querySelectorAll(selector);
+    if (!deleteButtons.length) return;
+
+    deleteButtons.forEach(button => {
+        button.addEventListener('click', async () => {
+            const passkeyName = button.getAttribute('data-passkey-name');
+            const passkeyId = button.getAttribute('data-passkey-id');
+            
+            if (!passkeyId) {
+                console.error('No passkey ID found');
+                alert('Error: Passkey ID not found');
+                return;
+            }
+            
+            if (confirm(`Are you sure you want to delete the passkey "${passkeyName}"?`)) {
+                try {
+                    const csrfToken = button.getAttribute('csrf-token');
+                    
+                    // Use the createAndSubmitForm function with passkeyId
+                    createAndSubmitForm('/passkey/delete', {
+                        passkeyId: passkeyId,
+                        csrfToken: csrfToken
+                    });
+                } catch (error) {
+                    console.error('Error deleting passkey:', error);
+                    alert('An error occurred while deleting the passkey');
```

Comment:
Should this be a toast?

Reply:
Fixed in https://github.com/guardian/janus-app/pull/571/commits/33e4b4d39bcbeaf63033bf3cd436a3337e633883